docs/operator-manual/rbac.md

```diff
@@ -203,9 +203,19 @@ p, example-user, applications, action/*, default/*, allow
 
 #### The `override` action
 
+The `override` action grant can be used to allow deploying arbitrary manifests or different revisions to existing Applications, e.g. for development or testing purposes. 
+
+**Attention:** This allows users to completely change/delete the deployed resources of the Application. 
+
 When granted along with the `sync` action, the override action will allow a user to synchronize local manifests to the Application.
 These manifests will be used instead of the configured source, until the next sync is performed.
 
+Additionally, when `application.sync.externalRevisionConsideredOverride: 'true'` is set in the `argcd-cm` configmap, 
+passing a revision when syncing an application is also considered as an `override`, to prevent synchronizing to arbitrary revisions other than the revision(s) given in the Application source.
+
+The default setting is 'false', to prevent breaking changes in existing installations. It is recommended to set this setting to 'true' and grant the `override` right per AppProject to the users that actually need this behavior.
```

Comment:
Should we set the default behavior to `false` (reverse the flag) so we are "secure" by default. Since the Sync action has to be called by a client, users will be able to upgrade Argo CD with the new behavior without affecting their Application. 

If users need tome to apply the override permissions, then they can disable the flag.